### DISCUSSION TITLE
How do I properly recreate authStore from just the token

### DISCUSSION THREAD

**Discussion by xrd** [pocketbase/pocketbase#2488] [Q&A] [answered]
I have a Firefox extension and want to provide the auth token to the pocketbase client (the firefox extension makes it more complicated to do a login with the integrated auth providers). So, right now I have the user login on a website (not inside the extension), and then get the token from `pb.authStore.token` then paste that into the extension popup dialog. I want the extension to then use that token and make authorized requests to the pocketbase backend.

I was hoping I could do this:

`pb.authStore.save(token)`.

But, it appears the authorization header is then not correct. If I inspect the headers using a `pb.beforeSend` hook I see an authorization header like:

```
Authorization: "eyJhbGciOi2398120398VCJ9"
```

And, I get errors like:  `"DrySubmit create rule failure: sql: no rows in result set` (which seems to be indicative that the API rules don't permit an anonymous user to create records).

When I watch the authorization header in a normal request it looks like a better JWT token.

```
Authorization: eyJhbGciOi2398120398VCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfYXV0aF82103982103htajMwcmRrNGN6Iiwid2392840ImF1dGhSZWNvcmQifQ.lK_ivte120982103GsZ6MCTzVI8bEMdwVkwc
```

I can decode that authorization token inside jwt.io and see that it has the user model serialized.

If I look at the normal user model, it appears to be like this:

```
 {                                                                                                                                                                            
    "avatar": "",                                                                                                                                                                       
    "collectionId": "_pb_users_auth_",                                                                                                                                                  
    "id": "1230981023",                                                                                                                                                            
    "collectionName": "users",                                                                                                                                                          
    "created": "2023-05-13 12:20:57.231Z",                                                                                                                                              
    "email": "foo@foo.com",                                                                                                                                                      
    "emailVisibility": false,                                                                                                                                                           
    "name": "",                                                                                                                                                                         
    "updated": "2023-05-13 12:20:57.231Z",                                                                                                                                              
    "username": "abc",                                                                                                                                                                  
    "verified": true,                                                                                                                                                                   
    "expand": {}                                                                                                                                                                        
  }  
```

Do I need to provide that to the `save` method? Is that then serialized into the jwt token inside the pocketbase JS library and sent as the authorization header? 

Is there a way to regenerate the model internally without doing that? I'm already asking a user to copy a token in, and asking them to copy in other fields from the model (say emaill and id, etc) would make it more error prone. I would love it if there were a way to refresh the user model just using the token, but I don't see anything to do that.

**Comment by ganigeorgiev** [comment, marked as answer]
> I was hoping I could do this: pb.authStore.save(token). But, it appears the authorization header is then not correct. 

The above is correct as long as `token` is a valid JWT auth token.

---

> I can decode that authorization token inside jwt.io and see that it has the user model serialized.

The auth token payload contains only the id and collectionId of the record. We don't serialize any other fields as they can change between requests.

---

> Do I need to provide that to the save method? Is that then serialized into the jwt token inside the pocketbase JS library and sent as the authorization header?

It is not required to provide an auth model when calling `pb.authStore.save()`. The token payload is generated on the server and as mentioned previously it contains only the minimum set of fields to identify the record.

---

>  I would love it if there were a way to refresh the user model just using the token, but I don't see anything to do that.

You can call `authRefresh()`:

```js
await pb.collection("users").authRefresh();
```

On success the above will automatically populate the store with the latest user model data. On missing, invalid or expired authStore token the above will throw an error.
